src/poetry/vcs/git/backend.py

```diff
@@ -342,7 +342,10 @@ def _clone_submodules(cls, repo: Repo) -> None:
                 source_root.mkdir(parents=True, exist_ok=True)
 
                 with repo:
-                    revision = repo.open_index()[path].sha.decode("utf-8")
+                    try:
+                        revision = repo.open_index()[path].sha.decode("utf-8")
+                    except KeyError:
```

Comment:
I'd add some `logger.debug()` message, just for ease of debugging this in the future.